src/Compilers/Core/Portable/Syntax/InternalSyntax/SyntaxDiagnosticInfoList.cs

```diff
@@ -1,11 +1,13 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace Microsoft.CodeAnalysis.Syntax.InternalSyntax
 {
     //This used to implement IEnumerable and have more functionality, but it was not tested or used.
-    internal struct SyntaxDiagnosticInfoList
+    internal struct SyntaxDiagnosticInfoList : IEnumerable<DiagnosticInfo>
```

Comment:
This is used here (and other places):
https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Test/Syntax/Scanner/ScannerTests.vb#L703

VB calls linq extension methods (i.e. Count) on the SyntaxDiagnosticInfoList